lib/ecto/migration.ex

```diff
@@ -700,29 +703,32 @@ defmodule Ecto.Migration do
   However, this feature does not work well with the transactions used by
   Ecto to guarantee integrity during migrations.
 
-  Therefore, to migrate indexes concurrently, you need to set
-  both `@disable_ddl_transaction` and `@disable_migration_lock` to true:
+  You can address this with two changes:
 
+    1. Change your repository to use PG advisory locks as the migration lock.
```

Comment:
@dbernheisel are there any downsides to using advisory locks? Only that they are new/experimental? We should add a note here.

Reply:
advisory locks certainly aren't new :) they've been in Postgres since v8.2.

The downsides is that they're less compatible and application managed, for example:

1. those that are using external connection pooling like pgBouncer won't be able to take advantage of advisory locks due to the way pgBouncer might share connections. pgBouncer uses session pool_mode by default which should work fine with advisory locks, but oftentimes admins will update it to transaction or statement pool_mode, which will start to get wonky when using session-level features such as advisory locks.

2. Postgres-compatible databases such as CockroachDB don't support advisory locks since they're built to be distributed.

3. There might be a connection failure between opening the lock and releasing the lock, leaving the database in a frozen state that cannot be migrated until the lock is released from the database. However, I tried testing a migration with `System.stop(1)` and exceptions in the middle of the migration or SIGINTs but the locks are released without issue likely due to the locks automatically releasing when the connection session is terminated. If the database (for some rare reason) thinks the session is still open yet the application is dead, that will be the problem area.

Having `:table_lock` be the default I think is great to avoid problems with points 1+2. Folks that are using pgBouncer and yet still select `:pg_advisory_lock` might find inconsistent behavior when obtaining and releasing the migration lock depending on their config. For this reason I'll added another exception for when releasing a lock fails so folks know this scenario happened. (FWIW, Rails also raises on failure to release the advisory migration lock) 

Those that choose `:pg_advisory_lock` may in rare cases still run into point 3, and there's nothing we can do about that other than wait for the servers' configured `tcp_keepalives_idle` to kick out of the dead session and therefore release the advisory lock, or restart the database.

I'll update the note here and add more detail in the Postgres adapter.